src/main/java/seedu/todo/commons/util/DateUtil.java

```diff
@@ -46,21 +54,44 @@ public static String formatDay(LocalDateTime dateTime) {
         LocalDate date = dateTime.toLocalDate();
         long daysDifference = LocalDate.now().until(date, ChronoUnit.DAYS);
 
-        String fromNow = "later";
-        String tillNow = "ago";
-
         // Consider today's date.
-        if (date.isEqual(LocalDate.now()))
-            return "Today";
+        if (date.isEqual(LocalDate.now())) {
+            return TODAY + SPACE + formatDateDisplay(date, true);
+        }
+        
+        if (daysDifference == 1) {
+            return TOMORROW + SPACE + formatDateDisplay(date, true);
+        }
 
         // Consider dates up to 6 days from today.
-        if (daysDifference > 0 && daysDifference <= 6)
-            return date.getDayOfWeek().getDisplayName(TextStyle.FULL, Locale.US);
+        if (daysDifference > 1 && daysDifference <= 6) {
+            return date.getDayOfWeek().getDisplayName(TextStyle.FULL, Locale.US) + SPACE + 
+                    formatDateDisplay(date, false);
+        }
 
         // Otherwise, dates should be a relative days ago/from now format.
         return String.format("%d %s %s", Math.abs(daysDifference), 
-                StringUtil.pluralizer((int) Math.abs(daysDifference), "day", "days"), 
-                daysDifference > 0 ? fromNow : tillNow);
+                StringUtil.pluralizer((int) Math.abs(daysDifference), DAY, DAYS), 
+                daysDifference > 0 ? FROM_NOW : TILL_NOW);
+    }
+    
+    /**
+     * Formats a LocalDateTime to a shorten date. 
+     * 
+     * @param dateTime   LocalDateTime to format, withDaysOfWeek.
+     * @return           Formatted shorten day.
+     */
+    private static String formatDateDisplay(LocalDate date, boolean withDaysOfWeek) {
+        //return with the days of the week
+        if (withDaysOfWeek) {
+            return date.getDayOfWeek().getDisplayName(TextStyle.FULL, Locale.US).substring(0, 3) + SPACE + 
+                    date.getDayOfMonth() + SPACE + 
+                    date.getMonth().getDisplayName(TextStyle.FULL, Locale.US).substring(0, 3);
+        }
+        else {
```

Comment:
Also, use `String.format` in this method instead of having `SPACE` everywhere might be better I think.